src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -146,6 +148,12 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
                 }
 
                 _socket = connectTask.Result;
+                if (_socket == null || !_socket.Connected)
```

Comment:
If the _socket is non-null but isn't connected, should we dispose it and null out _socket, or do we need to keep _socket assigned for some reason?